File: src/main/java/br/com/kaykemendes/learningjpapostgres/Person/PersonRepository.java
package br.com.kaykemendes.learningjpapostgres.Person;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Repository
public interface PersonRepository extends JpaRepository<Person, Long> {

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    @Modifying
    @Query(value = "UPDATE person SET name = ?1, age = ?2, gender = ?3 WHERE id = ?4", nativeQuery = true)
    void update(String name, int age, String gender, Long id);
}

File: src/main/java/br/com/kaykemendes/learningjpapostgres/Person/PersonRepository.kt
package br.com.kaykemendes.learningjpapostgres.Person

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

interface PersonRepository : JpaRepository<Person, Long> {

    fun findByName(name : String): Person

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    @Modifying
    @Query(value = "UPDATE person SET name = ?1, age = ?2, gender = ?3 WHERE id = ?4", nativeQuery = true)
    fun update(name: String, age: Int, gender: String, id: Long)

}